packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -875,7 +875,42 @@ const schema: SchemaType<"Comments"> = {
     canRead: ['guests'],
     canUpdate: ['sunshineRegiment', 'admins'],
     canCreate: ['members', 'sunshineRegiment', 'admins'],
-  } 
+  },
+
+  doppelComments: {
+    // Resolved by doppelComments resolver
+    type: Array,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['admins'],
+    canUpdate: ['admins'],
+    hidden: true,
+  },
+
+  'doppelComments.$': {
+    type: String,
+  },
+
+  ownDoppelCommentVote: resolverOnlyField({
+    type: 'DoppelCommentVote',
+    graphQLtype: 'DoppelCommentVote',
+    canRead: ['guests'],
+    resolver: async (comment: DbComment, args: void, context: ResolverContext) => {
+      const { DoppelCommentVotes } = context;
+      return await DoppelCommentVotes.findOne({commentId: comment._id, userId: context.currentUser?._id});
```

Comment:
This should probably do an early return in the case where there is no `currentUser`

Reply:
This also needs a loader and/or SQL resolver. N+1-select in a resolver that runs on every comment is quite a lot of queries.